airgun/views/acs.py

```diff
@@ -0,0 +1,447 @@
+        edit_url_and_subpaths = Button(
+            locator='//button[contains(@aria-label, "edit-urls-pencil-edit")]'
+        )
+
+        @View.nested
+        class url_and_subpaths_stack_content(View):
+            """Class representing content of the URL and subpaths stack item."""
+
+            ROOT = '//div[@id="showUrlPaths"]'
+
+            url = Text('//dd[@aria-label="url_text_value"]')
+            subpaths = Text('//dd[@aria-label="subpaths_text_value"]')
+
+    @View.nested
+    class credentials_stack_item(View, AcsStackItem):
+        """
+        Class representing the Credentials stack item in the ACS drawer.
+        Present only if ACS is of type 'Custom' or 'RHUI'.
+        """
+
+        ROOT = (
+            '//div[normalize-space(.)="Credentials" '
+            'and contains(@class, "pf-c-expandable-section")]'
+        )
+
+        title = OUIAText('expandable-credentials-text')
+        edit_credentials = Button(
+            locator='//button[contains(@aria-label, "edit-credentials-pencil-edit")]'
+        )
+
+        @View.nested
+        class credentials_stack_content(View):
+            """Class representing content of the Credentials stack item."""
+
+            ROOT = '//div[@id="showCredentials"]'
+
+            verify_ssl = Text('//dd[@aria-label="verifySSL_value"]')
+            ssl_ca_certificate = Text('//dd[@aria-label="sslCaCert_value"]')
+            ssl_client_certificate = Text('//dd[@aria-label="sslClientCert_value"]')
+            ssl_client_key = Text('//dd[@aria-label="sslClientKey_value"]')
+            username = Text('//dd[@aria-label="username_value"]')
+            password = Text('//dd[@aria-label="password_value"]')
+
+    @View.nested
+    class products_stack_item(View, AcsStackItem):
+        """
+        Class representing the Products stack item in the ACS drawer.
+        Present only if ACS is of type 'Simplified'.
+        """
+
+        ROOT = (
+            '//div[normalize-space(.)="Products" and contains(@class, "pf-c-expandable-section")]'
+        )
+
+        title = OUIAText('expandable-products-text')
+        edit_products = Button(
+            locator='//button[contains(@aria-label, "edit-products-pencil-edit")]'
+        )
+
+        @View.nested
+        class products_stack_content(View):
+            """Class representing content of the Products stack item."""
+
+            ROOT = '//div[@id="showProducts"]'
+
+            products_list = ItemsList(locator='.//ul[contains(@class, "pf-c-list")]')
+
+    def read(self):
+        """Reads the ACS drawer and returns its content as a dictionary."""
+
+        keys = ['details', 'capsules', 'url_and_subpaths', 'credentials', 'products']
+        result = dict.fromkeys(keys)
+        # Sleep ensures that all stack items are loaded and test does not fail
+        time.sleep(3)
+        # Expand and read each stack item
+        for key in result:
+            if key == 'details':
```

Comment:
you can avoid this completely if the naming of the item widget is the same as the key here.